pyccel/codegen/printing/fcode.py

```diff
@@ -659,8 +659,12 @@ def _print_NumpyNorm(self, expr):
         """Fortran print."""
         arg = PythonAbs(expr.arg) if isinstance(expr.arg.dtype, NativeComplex) else expr.arg
         if expr.axis:
-            axis = str(abs(expr.axis.python_value - arg.rank)) + '_C_DOUBLE'
-            code = 'Norm2({},{})'.format(self._print(arg), axis)
+            axis = expr.axis
+            if arg.order != 'F':
+                axis = PythonAbs(PyccelMinus(expr.axis, LiteralInteger(arg.rank)))
```

Comment:
Isn't `axis < rank`? Why not just do `PyccelMinus(LiteralInteger(arg.rank), expr.axis)` to avoid the abs?